clanker-deployer/DeployToken.ts

```diff
@@ -0,0 +1,407 @@
+    throw new Error(
+      `Vesting duration is too short. Minimum duration is ${MIN_VESTING_DURATION} seconds.`
+    );
+  }
+
+  // Helper to validate addresses
+  const validateAddress = (
+    address: string | undefined,
+    defaultAddress: `0x${string}`
+  ): `0x${string}` => {
+    if (!address || address.trim() === '' || !address.startsWith('0x')) {
+      return defaultAddress;
+    }
+    return address as `0x${string}`;
+  };
+
+  // Build the deployment config
+  const deploymentConfig = {
+    tokenConfig: {
+      name: formData.name,
+      symbol: formData.symbol,
+      salt: salt,
+      image: formData.imageUrl || '',
+      metadata: metadata,
+      context: socialContext,
+      originatingChainId: BigInt(chainId),
+    },
+    poolConfig: {
+      pairedToken: pairAddress,
+      tickIfToken0IsNewToken: initialTick || 0,
+    },
+    initialBuyConfig: {
+      pairedTokenPoolFee: 10000, // 10000 = 1%
+      pairedTokenSwapAmountOutMinimum: 0n,
+    },
+    vaultConfig: {
+      vaultDuration: vestingDuration,
+      vaultPercentage: formData.lockupPercentage || 0,
+    },
+    rewardsConfig: {
+      creatorReward: BigInt(Number(formData.creatorReward || 40)),
+      creatorAdmin: validateAddress(
+        formData.creatorRewardsAdmin,
+        deployerAddress
+      ),
+      creatorRewardRecipient: validateAddress(
+        formData.creatorRewardsRecipient,
+        deployerAddress
+      ),
+      interfaceAdmin: validateAddress(formData.interfaceAdmin, deployerAddress),
+      interfaceRewardRecipient: validateAddress(
+        formData.interfaceRewardRecipient,
+        deployerAddress
+      ),
+    },
+  } as const;
+
+  // Calculate value to send (dev buy amount)
+  const txValue =
+    typeof formData.devBuyAmount === 'number' && formData.devBuyAmount > 0
+      ? parseUnits(formData.devBuyAmount.toString(), 18)
+      : typeof formData.devBuyAmount === 'string' &&
+        formData.devBuyAmount !== ''
+      ? parseUnits(formData.devBuyAmount, 18)
+      : BigInt(0);
+
+  try {
+    // Encode the function call
+    const deployCalldata = encodeFunctionData({
+      abi: Clanker_v3_1_abi,
+      functionName: 'deployToken',
+      args: [deploymentConfig],
+    });
+
+    return {
+      transaction: {
+        to: CLANKER_FACTORY_V3_1,
+        data: deployCalldata,
+      },
+      value: txValue,
+    };
+  } catch (error: any) {
+    console.error('Error encoding function data:', error);
+    console.error('Problematic deployArgs:', deploymentConfig);
+    throw new Error(`Failed to encode function data: ${error.message}`);
+  }
+}
+
+/**
+ * Deploys a token using the Clanker contracts
+ */
+export async function deployToken({
+  deployerAddress,
+  formData,
+  chainId,
+  userInfo,
+  // Default desired price of 10 means:
+  // If your token has the lower address: 1 token = 0.1 WETH
+  // If paired token has the lower address: 1 WETH = 10 tokens
+  desiredPrice = 10,
```

Comment:
this is confusing that we don't internalize this lower/higher behavior like we do in the contracts

Reply:
can you just keep the 'Default price of 10 means: 1 token = 0.1 WETH'